schema-bridge.graphql

```diff
@@ -428,15 +897,338 @@ type PoolHourlySnapshot @entity {
+  to: Bytes!
+
+  " Address that sent the tokens "
+  from: Bytes!
+
+  " Pool where the tokens are deposited "
+  pool: Pool!
+
+  " Token deposited in the pool "
+  token: Token!
+
+  " Amount of token deposited "
+  amount: BigInt!
+
+  " Amount of token deposited in USD "
+  amountUSD: BigDecimal!
+
+  " Id of the chain where this is ocurring (should always be the chain where the subgraph is running) "
+  chainID: Int!
+
+  " Block number of this event "
+  blockNumber: BigInt!
+
+  " Timestamp of this event "
+  timestamp: BigInt!
+}
+
+type LiquidityWithdraw implements Event @entity(immutable: true) {
+  " event.transaction.hash.concatI32(event.logIndex) "
+  id: Bytes!
+
+  " Transaction hash of the transaction that emitted this event "
+  hash: Bytes!
+
+  " Event log index. For transactions that don't emit event, create arbitrary index starting from 0 "
+  logIndex: Int!
+
+  " The protocol this transaction belongs to "
+  protocol: BridgeProtocol!
+
+  " Account owner of this liquidity position. Should match the address in `to` "
+  account: Account!
+
+  " Address that received the tokens "
+  to: Bytes!
+
+  " Address that sent the tokens "
+  from: Bytes!
+
+  " Pool where the tokens are withdrawn from "
+  pool: Pool!
+
+  " Token withdrawn from the pool "
+  token: Token!
+
+  " Amount of token withdrawn "
+  amount: BigInt!
+
+  " Amount of token withdrawn in USD "
+  amountUSD: BigDecimal!
+
+  " Id of the chain where this is ocurring (should always be the chain where the subgraph is running) "
+  chainID: Int!
+
+  " Block number of this event "
+  blockNumber: BigInt!
+
+  " Timestamp of this event "
+  timestamp: BigInt!
+}
+
+enum TransferType {
+  MINT
+  BURN
+  LOCK
+  RELEASE
+}
+
+type BridgeTransfer implements Event @entity(immutable: true) {
+  " event.transaction.hash.concatI32(event.logIndex) "
+  id: Bytes!
+
+  " Transaction hash of the transaction that emitted this event "
+  hash: Bytes!
+
+  " Event log index. For transactions that don't emit event, create arbitrary index starting from 0 "
+  logIndex: Int!
+
+  " The protocol this transaction belongs to "
+  protocol: BridgeProtocol!
+
+  " Account that sent or received the transfer. Should match with either transferFrom or transferTo "
+  account: Account!
+
+  " Address that received the tokens "
+  to: Bytes!
+
+  " Address that sent the tokens "
+  from: Bytes!
+
+  " Are we briding in or out? "
+  isOutgoing: Boolean!
+
+  " Chain ID from where the transfer is being sent "
+  fromChainID: Int!
+
+  " Chain ID where the transfer is being sent to "
+  toChainID: Int!
+
+  " This is the address that's receiving the bridged funds (not a pool contract). When bridging out, it will be the address on the destination chain. "
+  transferTo: Bytes!
+
+  " This is the address initiating the transfer, where the initial funds came from. It might not match the address initiating the actual EVM transaction. "
+  transferFrom: Bytes!
```

Comment:
For multichain, I do not get `transferFrom` from the bridge-in event params, can we make this optional?

Reply:
Hmm, that makes sense 🤔 

Although, does multichain allow to send the transaction to a different address? Maybe it’s the same as the sender?